uamqp/client.py

```diff
@@ -282,16 +282,39 @@ def do_work(self, **kwargs):
             return True
         return self._client_run(**kwargs)
 
-    def mgmt_request(self, message, operation=None, operation_type=None, node='$management', **kwargs):
+    def mgmt_request(self, message, **kwargs):
         """
-        TODO: Move optional params to kwargs and document.
+        :param message: The message to send in the management request.
+        :type message: ~uamqp.message.Message
+        :keyword str operation: The type of operation to be performed. This value will
+         be service-specific, but common values include READ, CREATE and UPDATE.
+         This value will be added as an application property on the message.
+        :keyword str operation_type: The type on which to carry out the operation. This will
+         be specific to the entities of the service. This value will be added as
+         an application property on the message.
+        :keyword str node: The target node. Default node is `$management`.
+        :keyword float timeout: Provide an optional timeout in seconds within which a response
+         to the management request must be received.
+        :keyword callback: The function to process the returned parameters of the management
```

Comment:
Why do we take a callback instead of just returning the response and raising on an error code?

Reply:
This is carried over from [uamqp](https://github.com/Azure/azure-uamqp-python/blob/master/uamqp/client.py#L322) which takes the optional callback to parse the response.

In current service bus, callbacks are defined and passed when calling mgmt request.

I don't have strong preference here, is it that azure-core is doing the way you described? it makes sense to me if we want to align with azure-core implementation.

Reply:
I'd prefer to remove this callback - I can't see what it actually achieves as it's just being called before returning....
Additionally, the docstring should document the return type - do you know what `response` is that we're returning? Is that just the str?
Seems like we could just be returning a Tuple[int, str].

Reply:
the response is an amqp message with the body containing the content of the response and app_prop containg response code.

<img width="956" alt="image" src="https://user-images.githubusercontent.com/47871814/146592401-4b319c50-e54e-4ed7-8161-13606c3ffb96.png">